applications/GeoMechanicsApplication/custom_utilities/linear_nodal_extrapolator.cpp

```diff
@@ -13,54 +13,65 @@
 
 #include "linear_nodal_extrapolator.h"
 
+#include "custom_elements/interface_element.h"
+#include "element_utilities.hpp"
 #include "geometries/hexahedra_3d_8.h"
 #include "geometries/line_2d_2.h"
 #include "geometries/quadrilateral_2d_4.h"
 #include "geometries/tetrahedra_3d_4.h"
 #include "geometries/triangle_2d_3.h"
+#include "includes/element.h"
 #include "node_utilities.h"
 
 namespace Kratos
 {
 
-Matrix LinearNodalExtrapolator::CalculateElementExtrapolationMatrix(const GeometryType& rGeometry,
-                                                                    const GeometryData::IntegrationMethod& rIntegrationMethod) const
+Matrix LinearNodalExtrapolator::CalculateElementExtrapolationMatrix(const Element& rElement) const
 {
-    CheckIfGeometryIsSupported(rGeometry);
+    auto        p_interface_element = dynamic_cast<const InterfaceElement*>(&rElement);
+    const auto& r_geometry =
+        p_interface_element ? p_interface_element->GetMidGeometry() : rElement.GetGeometry();
+    CheckIfGeometryIsSupported(r_geometry);
 
-    const auto p_lower_order_geometry = CreateLowerOrderGeometry(rGeometry);
-    const GeometryType& p_corner_geometry = p_lower_order_geometry ? *p_lower_order_geometry : rGeometry;
+    const auto  p_lower_order_geometry = CreateLowerOrderGeometry(r_geometry);
+    const auto& r_corner_geometry = p_lower_order_geometry ? *p_lower_order_geometry : r_geometry;
 
-    Matrix extrapolation_matrix =
-        CalculateExtrapolationMatrixForCornerNodes(rGeometry, rIntegrationMethod, p_corner_geometry);
+    const auto integration_points = GeoElementUtilities::GetIntegrationPointsOf(rElement);
 
-    if (p_lower_order_geometry) {
-        AddRowsForMidsideNodes(rGeometry, extrapolation_matrix);
+    auto result = CalculateExtrapolationMatrixForCornerNodes(r_geometry, integration_points, r_corner_geometry);
+
+    if (p_lower_order_geometry) AddRowsForMidsideNodes(r_geometry, result);
+
+    if (p_interface_element) {
+        // The extrapolation matrix has been calculated for the mid-geometry. Now we still need to
+        // expand to it the full interface geometry (which comprises two sides).
+        auto tmp = Matrix{2 * result.size1(), result.size2(), 0.0};
+        GeoElementUtilities::AddMatrixAtPosition(result, tmp, 0, 0);
+        GeoElementUtilities::AddMatrixAtPosition(result, tmp, result.size1(), 0);
+        std::swap(result, tmp);
```

Comment:
I think this can be simplified a bit (which also reduces the number of operations), since `resize` by default preserves the data (unless you specify preserve = false when calling the function)
```suggestion
        result.resize(2 * result.size1(), result.size2());
        GeoElementUtilities::AddMatrixAtPosition(result, tmp, result.size1(), 0);
```

Reply:
Agreed, this is simpler, and perhaps it also performs slightly better (I didn't bother to measure it though). To make your suggestion a little easier to understand (due to changing number of rows, before and after resizing the matrix), I've introduced a constant named `number_of_rows_per_side`. Hope you agree with that tiny change of your suggestion.

Reply:
Yes, that made it more readable indeed 👍